### PR TITLE
Using subCategory field for GSI to have filtering based of type of index

### PR DESCRIPTION
**Pull request by mahesh152**

### CODE REVIEW

File: app/menu.json [hunk 1]
```diff
@@ -76,32 +76,19 @@
...
-				"id": "plasma_thr", "title": "Throughput PLASMA"
-			}, {
-				"id": "initial_plasma", "title": "Initial PLASMA"
+				"id": "recovery", "title": "Recovery"
 			}, {
-				"id": "incremental_plasma", "title": "Incremental PLASMA"
-			}]
+				"id": "standalone", "title": "Standalone"
+			}],
+			"sub_categories": ["moi", "plasma", "fdb"]
```
Comment (pavel-paulau) [inline app/menu.json]: Maybe we should use more user-friendly sub-categories. E.g., MOI, Plasma, FDB or ForestDB.

File: datasources.go [hunk 2]
```diff
@@ -114,14 +114,14 @@ func (ds *dataStore) addMetric(metric Metric) error {
 	return err
 }
 
-func (ds *dataStore) getMetrics(component, category string) (*[]MetricWithCluster, error) {
+func (ds *dataStore) getMetrics(component, category string, sub_category string) (*[]MetricWithCluster, error) {
```
Comment (pavel-paulau) [inline datasources.go]: sub_category -> subCategory

File: datasources.go [hunk 3]
```diff
@@ -114,14 +114,14 @@ func (ds *dataStore) addMetric(metric Metric) error {
...
 	var metrics []MetricWithCluster
 
 	query := gocb.NewN1qlQuery(
-		"SELECT m.id, m.title, m.component, m.category, m.orderBy, c AS `cluster` " +
+		"SELECT m.id, m.title, m.component, m.category, m.orderBy, m.subCategory, c AS `cluster` " +
 			"FROM metrics m JOIN clusters c ON KEYS m.`cluster`" +
-			"WHERE m.component = $1 AND m.category = $2;")
-	params := []interface{}{component, category}
+			"WHERE m.component = $1 AND m.category = $2 AND m.subCategory = $3;")
```
Comment (pavel-paulau) [inline datasources.go]: Please also create a new index for this query.

File: datasources.go [hunk 4]
```diff
@@ -220,17 +220,17 @@ func (ds *dataStore) addBenchmark(benchmark Benchmark) error {
 	return err
 }
 
-func (ds *dataStore) getBenchmarks(component, category string) (*[]Benchmark, error) {
+func (ds *dataStore) getBenchmarks(component, category string, sub_category string) (*[]Benchmark, error) {
```
Comment (pavel-paulau) [inline datasources.go]: sub_category -> subCategory

File: datasources.go [hunk 5]
```diff
@@ -220,17 +220,17 @@ func (ds *dataStore) addBenchmark(benchmark Benchmark) error {
...
 	benchmarks := []Benchmark{}
 
 	query := gocb.NewN1qlQuery(
 		"SELECT b.`build`, b.id, b.hidden, b.metric, b.`value` " +
 			"FROM metrics m " +
 			"JOIN benchmarks b " +
 			"ON KEY b.metric FOR m " +
-			"WHERE m.component = $1 AND m.category = $2 " +
+			"WHERE m.component = $1 AND m.category = $2 AND m.subCategory = $3" +
```
Comment (pavel-paulau) [inline datasources.go]: Please also create a new index for this query.

File: rest.go [hunk 6]
```diff
@@ -32,8 +32,12 @@ func addBenchmark(c *gin.Context) {
 func getBenchmarks(c *gin.Context) {
 	component := c.Param("component")
 	category := c.Param("category")
+	sub_category := c.Param("sub_category")
```
Comment (pavel-paulau) [inline rest.go]: sub_category -> subCategory

File: rest.go [hunk 7]
```diff
@@ -161,14 +169,14 @@ func httpEngine() *gin.Engine {
 	rg.GET("builds", getBuilds)
 
 	rg.POST("benchmarks", addBenchmark)
-	rg.GET("benchmarks/:component/:category", getBenchmarks)
+	rg.GET("benchmarks/:component/:category/:sub_category", getBenchmarks)
```
Comment (pavel-paulau) [inline rest.go]: sub_category -> subCategory

File: app/js/timeline.js [hunk 8]
```diff
@@ -3,9 +3,9 @@ angular
 	.config([
 		'$routeProvider', function($routeProvider) {
 			$routeProvider.
-				when('/timeline/:os/:component/:category', {templateUrl: '/static/timeline.html', controller: MenuRouter}).
+				when('/timeline/:os/:component/:category/:sub_category', {templateUrl: '/static/timeline.html', controller: MenuRouter}).
```
Comment (pavel-paulau) [inline app/js/timeline.js]: sub_category -> subCategory

File: app/menu.json (hunk 1 above)
Comment (pavel-paulau) [inline app/menu.json]: sub_categories -> subCategories

File: app/js/admin.js [hunk 9]
```diff
@@ -13,6 +13,7 @@ function InitMenu($scope, $http) {
 		$scope.components = menu.components;
 		$scope.activeComponent = Object.keys($scope.components)[0];
 		$scope.activeCategory = $scope.components[$scope.activeComponent].categories[0].id;
+		$scope.activeSubCategory = $scope.components[$scope.activeComponent].sub_categories[0];
```
Comment (pavel-paulau) [inline app/js/admin.js]: Why do you need to make changes to the administration dashboard?

File: app/js/timeline.js [hunk 10]
```diff
@@ -75,6 +90,13 @@ function DefineFilters($scope) {
 				return metric.cluster.os.indexOf("Windows") === 0;
 		}
 	};
+
+	$scope.bySubCategory = function(metric) {
+		if ($scope.activeSubCategory == "all") {
+			return metric.subCategory == "";
```
Comment (pavel-paulau) [inline app/js/timeline.js]: return true?

File: app/js/admin.js (hunk 9 above)
Comment (mahesh152) [inline app/js/admin.js]: Ok.

File: app/js/timeline.js (hunk 10 above)
Comment (mahesh152) [inline app/js/timeline.js]: I am removing this as, as we are not fetching all records every time.

File: app/js/timeline.js (hunk 8 above)
Comment (mahesh152) [inline app/js/timeline.js]: Ok.

File: rest.go (hunk 7 above)
Comment (mahesh152) [inline rest.go]: Ok.

File: rest.go (hunk 6 above)
Comment (mahesh152) [inline rest.go]: Ok.

File: datasources.go (hunk 5 above)
Comment (mahesh152) [inline datasources.go]: Ok.

File: datasources.go (hunk 4 above)
Comment (mahesh152) [inline datasources.go]: Ok.

File: datasources.go (hunk 3 above)
Comment (mahesh152) [inline datasources.go]: Ok.

File: datasources.go (hunk 2 above)
Comment (mahesh152) [inline datasources.go]: Ok.

File: app/menu.json (hunk 1 above)
Comment (mahesh152) [inline app/menu.json]: Ok.

Comment (mahesh152) [inline app/menu.json]: Ok.

**Pull request closed without merging**